external/types/web.d.ts

```diff
@@ -0,0 +1,992 @@
+     * @returns {AnnotationLayerBuilder}
+     */
+    createAnnotationLayerBuilder(
+      pageDiv: HTMLDivElement,
+      pdfPage: PDFPageProxy,
+      imageResourcesPath?: string,
+      renderInteractiveForms?: boolean,
+      l10n?: IL10n
+    ): AnnotationLayerBuilder;
+  }
+
+  interface IPDFLinkService {
+    /**
+     * @returns {number}
+     */
+    page: number;
+
+    /**
+     * @returns {number}
+     */
+    rotation: number;
+
+    /**
+     * @param {string|Array} dest - The named, or explicit, PDF destination.
+     */
+    navigateTo(dest: Destination): void;
+
+    /**
+     * @param {string|Array} dest - The PDF destination object.
+     * @returns {string} The hyperlink to the PDF object.
+     */
+    getDestinationHash(dest: Destination): string;
+
+    /**
+     * Prefix the full url on anchor links to make sure that links are resolved
+     * relative to the current URL instead of the one defined in <base href>.
+     * @param {String} anchor The anchor hash, including the #.
+     * @returns {string} The hyperlink to the PDF object.
+     */
+    getAnchorUrl(hash: string): string;
+
+    /**
+     * @param {string} hash
+     */
+    setHash(hash: string): void;
+
+    /**
+     * @param {string} action
+     */
+    executeNamedAction(action: string): void;
+
+    /**
+     * @param {Object} params
+     */
+    onFileAttachmentAnnotation(params: {
+      id: string;
+      filename: string;
+      content: any;
+    }): void;
+
+    /**
+     * @param {number} pageNum - page number.
+     * @param {Object} pageRef - reference to the page.
+     */
+    cachePageRef(pageNum: number, pageRef: Ref): void;
+  }
+
+  //
+  // web/pdf_link_service.js
+  //
+  interface SimpleLinkService extends IPDFLinkService {}
+  let SimpleLinkService: new () => SimpleLinkService;
+
+  interface PDFLinkService extends IPDFLinkService {
+    eventBus: EventBus;
+    baseUrl: string;
+    pdfDocument: PDFDocumentProxy;
+    pdfViewer: PDFViewer;
+    pdfHistory: PDFHistory;
+    setDocument(pdfDocument: PDFDocumentProxy, baseUrl: string): void;
+    setViewer(pdfViewer: PDFViewer): void;
+    setHistory(pdfHistory: PDFHistory): void;
+  }
+
+  /**
+   * @typedef {Object} PDFLinkServiceOptions
+   * @property {EventBus} eventBus - The application event bus.
+   * @property {number} externalLinkTarget - (optional) Specifies the `target`
+   *   attribute for external links. Must use one of the values from {LinkTarget}.
+   *   Defaults to using no target.
+   * @property {string} externalLinkRel - (optional) Specifies the `rel` attribute
+   *   for external links. Defaults to stripping the referrer.
+   */
+  interface PDFLinkServiceOptions {
+    eventBus: EventBus;
+    externalLinkTarget?: LinkTarget;
+    externalLinkRel?: string;
+  }
+
+  /**
+   * @param {PDFLinkServiceOptions} options
+   * Constructor for PDFLinkService
+   */
+  let PDFLinkService: new (options: PDFLinkServiceOptions) => PDFLinkService;
```

Comment:
I do not know how up-to-date this example is but it does not supply any parameter to the constructor. https://github.com/mozilla/pdf.js/blob/master/examples/components/simpleviewer.js

Reply:
You're right.